root/utility/formatUserDate.js

```diff
@@ -81,10 +104,22 @@ function formatDateUsingPattern(
   const cacheKey = pattern + '-' + timezone + '-' + locale;
   let formatter = formatterCache.get(cacheKey);
   if (!formatter) {
-    formatter = new Intl.DateTimeFormat(
-      locale,
-      Object.assign({timeZone: timezone}, options),
-    );
+    try {
+      formatter = new Intl.DateTimeFormat(
+        locale,
+        Object.assign({timeZone: timezone}, options),
+      );
+    } catch (e) {
+      if (e instanceof RangeError && /time zone/.test(e.message)) {
```

Comment:
In Node v12.12.0 on macOS I get a slightly different error message, "RangeError: Invalid time zone specified: CET," not "RangeError: Expected Area/Location(/Location)* for time zone, got CET" as on beta (where I believe we use Node v10). So I'm just checking for "time zone" in the message here. (I do believe there are other cases where a `RangeError` might be thrown that has nothing to do with the time zone, which is why I'm checking the message at all.)

Reply:
Also note that "CET" works fine in my browser (Firefox 70.0.1) despite not working in Node. It's implementation-dependent what time zone names are supported.

Reply:
Node.js supports UTC, IANA time zones (Area/City), and GMT offsets (GMT±X).